src/Datadog.Trace/Util/SpanIdGenerator.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+
+namespace Datadog.Trace.Util
+{
+    internal class SpanIdGenerator
+    {
+#if !NETSTANDARD
+        private static readonly Random GlobalSeedGenerator = new Random();
+#endif
+
+        [ThreadStatic]
+        private static Random _threadRandom;
+
+        public static ulong CreateNew()
+        {
+            Random random = GetRandom();
+
+            long high = random.Next(int.MinValue, int.MaxValue);
+            long low = random.Next(int.MinValue, int.MaxValue);
+
+            // Concatenate both values, and truncate the 32 top bits from low
+            var value = high << 32 | (low & 0xFFFFFFFF);
+
+            return (ulong)value & 0x7FFFFFFFFFFFFFFF;
+        }
+
+        private static Random GetRandom()
+        {
+            Random random = _threadRandom;
+
+            if (random == null)
+            {
+#if NETSTANDARD
+                random = new Random();
```

Comment:
Is there a reason to use a local instead of just setting `_threadRandom = new Random()` ?

Reply:
TLS reads are relatively costly. Storing the value in a local variable saves 1 read